src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -221,6 +221,11 @@ private static bool IsNamedTypeAccessible(NamedTypeSymbol type, Symbol within, r
 
             failedThroughTypeCheck = false;
 
+            if (containingType.IsTupleType)
+            {
+                containingType = containingType.TupleUnderlyingType;
```

Comment:
This unwrapping is required for proper accessibility checks when we are compiling code within ValueTuple<> declarations. I'll try to come up with some tests that would complain about inaccessible things in VT and will see if any adjustments are warranted. 

---
In reply to: [62271140](https://github.com/dotnet/roslyn/pull/11136#discussion_r62271140) [](ancestors = 62271140)